src/main/java/org/skriptlang/skript/bukkit/recipes/elements/CondDiscoveredRecipes.java

```diff
@@ -0,0 +1,75 @@
+package org.skriptlang.skript.bukkit.recipes.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.bukkitutil.NamespacedUtils;
+import ch.njol.util.Kleenean;
+import org.bukkit.Bukkit;
+import org.bukkit.Keyed;
+import org.bukkit.NamespacedKey;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.Recipe;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Has Discovered Recipe")
+@Description("Checks whether the players have discovered a recipe.")
+@Examples({
+	"if player has discovered recipe \"custom_recipe\":",
+		"\tgive player 1 diamond",
+	"",
+	"if all players have not discovered recipe \"custom_recipe\":",
+		"\tkill all players",
+})
+@Since("INSERT VERSION")
+public class CondDiscoveredRecipes extends Condition {
+
+	static {
+		Skript.registerCondition(CondDiscoveredRecipes.class,
+			"%players% (has|have) (discovered|unlocked) [the] recipe[s] %recipes/strings%",
+			"%players% (hasn't|has not|haven't|have not) (discovered|unlocked) [the] recipe[s] %recipes/strings%");
```

Comment:
Well I know I can do only `%recipes%` which then the strings couldbe used via ExprGetRecipe. But because of the syntax of both, it would then ben
`if player has discovered recipe (recipe with the key "my_recipe")`

Reply:
I don't really think that's an issue, but you could also just remove `recipe` from the condition's syntax.